### PR TITLE
Test/add test for use api cache

### PR DESCRIPTION
**Pull request by Jeffrey-Zutt**
Unit tests for API hooks

### CODE REVIEW

File: src/app/state/rest/hooks/useRequestQueue.tsx [hunk 1]
```diff
@@ -33,13 +33,13 @@ const reducer = produce((draft: State, action: Action) => {
   }
 })
 
-const isPending = (state: Readonly<State>, url: string): boolean =>
-  state.find(_ => _.url === url) !== undefined
+const isPending = (state: Readonly<State>, url: string, method: string): boolean =>
```
Comment (gilleswittenberg) [inline src/app/state/rest/hooks/useRequestQueue.tsx:36]: Could we use `type Method = "GET" | "POST" | "PUT" | "PATCH" | "DELETE"`?

File: src/app/state/rest/hooks/useApiRequest.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ type Config = {
   getHeaders?: () => Record<string, string>
 }
 
-type Callback = () => void
+export type Callback = () => void
```
Comment (gilleswittenberg) [inline src/app/state/rest/hooks/useApiRequest.ts:13]: This can't be inferred? Exporting is only for unit tests?

Comment (Jeffrey-Zutt) [reply src/app/state/rest/hooks/useApiRequest.ts:13]: Yeah. Good one.
Sharp eye you have ;-)

File: src/app/state/rest/hooks/useRequestQueue.tsx (hunk 1 above)
Comment (Jeffrey-Zutt) [reply src/app/state/rest/hooks/useRequestQueue.tsx:36]: Yes. I think method is already somewhere defined even.

**Pull request merged**